src/saveload/oldloader.cpp

```diff
@@ -134,8 +132,8 @@ bool LoadChunk(LoadgameState *ls, void *base, const OldChunks *chunks)
 						break;
 
 					case OC_ASSERT:
-						DEBUG(oldloader, 4, "Assert point: 0x%X / 0x%X", ls->total_read, chunk->offset + _bump_assert_value);
-						if (ls->total_read != chunk->offset + _bump_assert_value) throw std::exception();
+						DEBUG(oldloader, 4, "Assert point: 0x%X / 0x%X", ls->total_read, (uint)(size_t)chunk->ptr + _bump_assert_value);
+						if (ls->total_read != (size_t)chunk->ptr + _bump_assert_value) throw std::exception();
```

Comment:
Sorry for the most likely stupid question, but why isn't this something like `chunk->offset(_bump_assert_value)` or what-ever? It feels out-of-place compared to the rest. But honestly, my understanding of this change is limited, just validating this is correct :)

Reply:
Can't really get rid of `ptr` due to stuff like `OC_DEREFERENCE_POINTER` and how skips are indicated, so may  as well use it.